etcd/etcdtest/mocketcd.go

```diff
@@ -117,6 +135,12 @@ func Stop() error {
 		return nil
 	}
 
+	// Ensure that etcd's data-dir (etcdDataDir) is deleted even if killing the process fails.
+	defer func() {
+		os.RemoveAll(etcdDataDir)
+		etcdDataDir = "" // Clear the dataDir variable.
```

Comment:
IMO this and other comments are redundant as it is quite clear what is going on.

Reply:
I stripped the comments as supposed.